### ISSUE TITLE
The composer install --global drush/drush approach is deprecated (consider drush launcher)

### ISSUE THREAD

**Issue by rfay** [drud/ddev#1039]
**Describe the bug**

Over in http://docs.drush.org/en/8.x/install-alternative/ they basically *beg* us not to use the approach we're currently using in the web container's Dockerfile (`RUN composer global require drush/drush:${DRUSH_VERSION} && ln -s /home/.composer/vendor/drush/drush/drush /usr/local/bin/drush && chmod -R ugo+rx /root`

The new approach is to use the cgr tool.

Note, though:
* drush 8 is only good through Drupal 8.3 they say
* drush 9 is only good for Drupal 8.4 plus (although the site-local drush seems to work out here). 

So we may want to swap out the version of drush in /usr/local/bin based on the Drupal version.

Also note https://github.com/drud/ddev/issues/1028 and drush launcher, which may be the best path forward. But don't forget backdrop, which has requirements of its own.

**Comment by CoreyWycliffe** [comment]
just as a note, drush did not work in my ddev container `ddev ssh` in the `/var/www/html` folder until I downloaded the drush launcer. See instructions @ https://github.com/drush-ops/drush-launcher/blob/master/README.md#installation---phar

I'm on Windows 10
and was seeing this error

`PHP Warning:  pcntl_exec(): Error has occurred: (errno 8) Exec format error in /home/.composer/vendor/drush/drush/includes/startup.inc on line 415

Warning: pcntl_exec(): Error has occurred: (errno 8) Exec format error in /home/.composer/vendor/drush/drush/includes/startup.inc on line 415
Error has occurred executing the Drush script found at /var/www/html/vendor/bin/drush
(errno 8) Exec format error`

**Comment by rfay** [comment]
@CoreyWycliffe I would imagine your docroot is not set correctly. Could you please check to make sure that it's set to the same directory where your index.php is? I use drush successfully all the time, just tested it on Windows 10 Pro/Docker-for-Windows, and it should be fine.

Drush never has been intended to work in drupal unless in the docroot or the --root is provided. Is that's what's going on?

Basically if the docroot is set correctly in .ddev/config.yaml, `ddev ssh` will land you in the docroot on Drupal. And drush will work there. Make sure, of course, that you have done your composer install (otherwise you get problems like this) and you have a database loaded or installed.

**Comment by rfay** [comment]
Oh, I note that it's having trouble running vendor/bin/drush, which is sometimes a symlink, and that can be the issue. ddev v1.5+ has special handling for symlinks in `ddev composer` *if* you're running docker-for-windows and  have developer mode enabled in Win10 Pro.

**Comment by rfay** [comment]
And... if you don't run composer inside the container you'll hit problems of this type, because the symlink created on the host tries to get pushed into the container, where it doesn't work. We recommend using `ddev composer install`, which tries to make all that work out for you by doing everything inside the container and avoiding filesystem differences between NTFS and the filesystem inside docker.

**Comment by CoreyWycliffe** [comment]
Sorry for the delayed response. Yea, I did use ddev 1.5 with the same results. My docroot is set to "", all the files are in the main directory. `ddev `does run and will produce a website. It's just that going into the container results in not being able to run `drush`. The drupal site that I'm using does not use the typical composer template, it's just a directory with the drupal code. I'm trying to give you all the information I know.

**Comment by rfay** [comment]
Happy to help you with this, but it needs to be in a different issue or on Drupal slack #ddev channel.  I imagine the problem is with your composer setup.
